features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -425,6 +433,23 @@ public Component getComponent() {
         });
     }
 
+    String decodeResourceId(String resourceId){
+        // This is a quite ugly hack but it will be removed as of foundation-2018 when the class ResourceId will be
```

Comment:
Yeah, this URL encoding caused a number of other issues too. Glad to see the ResourceId class help mitigate this.